### ISSUE TITLE
Support for new NodeIP configuration option

### ISSUE THREAD

**Issue by AndrewArmitage** [hashicorp/terraform-provider-azurerm#20161]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The az cli tools allows AKS clusters to be created using the new NodeIP setting for the load balancers; https://learn.microsoft.com/en-us/azure/aks/load-balancer-standard#change-the-inbound-pool-type-preview

az aks create \
    --resource-group myResourceGroup \
    --name myAKSCluster \
    --load-balancer-backend-pool-type=nodeIP

It would be good if the terraform provider allowed setting that same option.

### New or Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Potential Terraform Configuration

```hcl
load_balancer_profile {
  backend_pool_type = nodeIP
}
```

**Comment by ericwil-git** [comment]
I completely agree! We need to get the Terraform provider updated to include this new setting! Good call-out!
